Mathlib/Data/Set/Prod.lean

```diff
@@ -926,3 +934,29 @@ theorem univ_pi_ite (s : Set ι) [DecidablePred (· ∈ s)] (t : ∀ i, Set (α
 end Pi
 
 end Set
+
+namespace Equiv
+
+open Set
+variable {ι ι' : Type*} {α : ι → Type*}
+
+theorem piCongrLeft_symm_preimage_pi (f : ι' ≃ ι) (s : Set ι) (t : ∀ i, Set (α i)) :
+    ((f.piCongrLeft α).symm ⁻¹' (f ⁻¹' s).pi fun i' => t <| f i') = s.pi t := by
```

Comment:
Assuming it's still true, this seems like a more useful lemma to me:
```suggestion
    ((f.piCongrLeft α).symm ⁻¹' s.pi fun i' => t <| f i') = (f.symm ⁻¹' s).pi t := by
```

Reply:
I agree